control-plane/connect-inject/controllers/endpointsv2/endpoints_controller.go

```diff
@@ -202,23 +203,24 @@ func getOwnerPrefixFromPod(pod *corev1.Pod) string {
 
 // registerService creates a Consul service registration from the provided Kuberetes service and endpoint information.
 func (r *Controller) registerService(ctx context.Context, resourceClient pbresource.ResourceServiceClient, service corev1.Service, selector *pbcatalog.WorkloadSelector) error {
-	serviceResource := r.getServiceResource(
-		&pbcatalog.Service{
-			Workloads:  selector,
-			Ports:      getServicePorts(service),
-			VirtualIps: r.getServiceVIPs(service),
-		},
+	consulSvc := &pbcatalog.Service{
+		Workloads:  selector,
+		Ports:      getServicePorts(service),
+		VirtualIps: r.getServiceVIPs(service),
+	}
+	consulSvcResource := r.getServiceResource(
+		consulSvc,
 		service.Name, // Consul and Kubernetes service name will always match
 		r.getConsulNamespace(service.Namespace),
 		r.getConsulPartition(),
 		getServiceMeta(service),
 	)
 
-	r.Log.Info("registering service with Consul", getLogFieldsForResource(serviceResource.Id)...)
+	r.Log.Info("registering service with Consul", getLogFieldsForResource(consulSvcResource.Id)...)
 	//TODO: Maybe attempt to debounce redundant writes. For now, we blindly rewrite state on each reconcile.
-	_, err := resourceClient.Write(ctx, &pbresource.WriteRequest{Resource: serviceResource})
+	_, err := resourceClient.Write(ctx, &pbresource.WriteRequest{Resource: consulSvcResource})
 	if err != nil {
-		r.Log.Error(err, "failed to register service", getLogFieldsForResource(serviceResource.Id)...)
+		r.Log.Error(err, fmt.Sprintf("failed to register service: %+v", consulSvc), getLogFieldsForResource(consulSvcResource.Id)...)
```

Comment:
This logging seemed valuable to keep around. There's no secrets in this value, so I think it's reasonable to include. Happy to remove if folks disagree